### PR TITLE
Bigc 618 measure bigcommerce api response

### PR DESCRIPTION
**Pull request by aaron-martin**

### CODE REVIEW

File: extension/lib/catalog/product/repository/ShopgateProductListRepository.js [hunk 1]
```diff
@@ -29,8 +31,9 @@ class ShopgateProductListRepository {
     /**
      * @type BigCommercePage
      */
+    this._storeLogger.startTimmer()
```
Comment (Menes1337) [inline extension/lib/catalog/product/repository/ShopgateProductListRepository.js]: startTimer I guess

File: test/integration/extension/lib/catalog/product/repository/ShopgateDescriptionRepositoryTest.js [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
+const StoreLogger = require('../../../../../../../extension/lib/store/logger/StoreLogger')
 
 const factory = new BigCommerceFactory(Credentials.clientId, Credentials.accessToken, Credentials.storeHash)
 
 describe('Product description by product id', () => {
+  const context = {
+    log: {
+      info: function(message) {
+
+      }
```
Comment (Menes1337) [inline test/integration/extension/lib/catalog/product/repository/ShopgateDescriptionRepositoryTest.js]: to be consistent with the other context objects in the other test cases put the brackets in one line: info: function(message) {}

File: extension/lib/catalog/product/repository/ShopgateProductListRepository.js (hunk 1 above)
Comment (aaron-martin) [reply extension/lib/catalog/product/repository/ShopgateProductListRepository.js]: corrected

File: test/integration/extension/lib/catalog/product/repository/ShopgateDescriptionRepositoryTest.js (hunk 2 above)
Comment (aaron-martin) [reply test/integration/extension/lib/catalog/product/repository/ShopgateDescriptionRepositoryTest.js]: corrected

File: extension/lib/store/logger/StoreLogger.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  _createTimeLongMetaObject (startTime, endTime, description) {
+    let duration = endTime - startTime
+    return {duration: duration, description: description}
+  }
+
+  startTimmer () {
+    this._startTime = new Date()
+  }
+
+  logTime (description) {
```
[the remark this answers is not in the document]
Comment (aaron-martin) [reply extension/lib/store/logger/StoreLogger.js]: corrected

File: extension/lib/store/logger/StoreLogger.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  }
+
+  /**
+   * @param {Date} startTime
+   * @param {Date} endTime
+   * @param {string} description
+   * @return {TimeLogMetaData}
+   */
+  _createTimeLongMetaObject (startTime, endTime, description) {
+    let duration = endTime - startTime
```
[the remark this answers is not in the document]
Comment (aaron-martin) [reply extension/lib/store/logger/StoreLogger.js]: corrected

File: extension/lib/store/logger/StoreLogger.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+class StoreLogger {
+  constructor (context) {
```
[the remark this answers is not in the document]
Comment (aaron-martin) [reply extension/lib/store/logger/StoreLogger.js]: corrected but there is an issue with the documentation of the info method of context.log. I will correct that now.

[the remark this answers is not in the document]
Comment (aaron-martin) [reply extension/lib/store/logger/StoreLogger.js]: removed unneeded and incorrect type hinting ofr context.log.info method

[the remark this answers is not in the document]
Comment (Menes1337) [reply extension/lib/store/logger/StoreLogger.js]: @aaron-martin I guess you should rename all Parameter types from LoggerContextLog to LoggerContext

**Pull request merged**